examples/camera/2d_screen_shake.rs

```diff
@@ -1,70 +1,201 @@
+    sprite::MeshMaterial2d,
+};
+
+// Before we implement the code, let's quickly introduce the underlying constants.
+// They are later encoded in a `CameraShakeConfig` component, but introduced here so we can easily tweak them.
+// Try playing around with them and see how the shake behaves!
 
-use bevy::{prelude::*, render::camera::SubCameraView, sprite::MeshMaterial2d};
-use rand::{Rng, SeedableRng};
-use rand_chacha::ChaCha8Rng;
+/// The trauma decay rate controls how quickly the trauma decays.
+/// 0.5 means that a full trauma of 1.0 will decay to 0.0 in 2 seconds.
+const TRAUMA_DECAY_PER_SECOND: f32 = 0.5;
 
-const CAMERA_DECAY_RATE: f32 = 0.9; // Adjust this for smoother or snappier decay
-const TRAUMA_DECAY_SPEED: f32 = 0.5; // How fast trauma decays
-const TRAUMA_INCREMENT: f32 = 1.0; // Increment of trauma per frame when holding space
+/// The trauma exponent controls how the trauma affects the shake.
+/// Camera shakes don't feel punchy when they go up linearly, so we use an exponent of 2.0.
+/// The higher the exponent, the more sudden the shake kicks in.
+const TRAUMA_EXPONENT: f32 = 2.0;
 
-// screen_shake parameters, maximum addition by frame not actual maximum overall values
-const MAX_ANGLE: f32 = 0.5;
-const MAX_OFFSET: f32 = 500.0;
+/// The maximum angle the camera can rotate on full trauma.
+/// 10.0 degrees is a somewhat high but still reasonable shake. Try bigger values for something more silly and wiggly.
+const MAX_ANGLE: f32 = 10.0_f32.to_radians();
 
-#[derive(Component)]
-struct Player;
+/// The maximum translation the camera will move on full trauma in both the x and y directions.
+/// 20.0 px is a low enough displacement to not be distracting. Try higher values for an effect that looks like the camera is wandering around.
+const MAX_TRANSLATION: f32 = 20.0;
+
+/// How much we are traversing the noise function in arbitrary units per second.
+/// This dictates how fast the camera shakes.
+/// 20.0 is a fairly fast shake. Try lower values for a more dreamy effect.
+const NOISE_SPEED: f32 = 20.0;
+
+/// How much trauma we add per press of the space key.
+/// A value of 1.0 would mean that a single press would result in a maximum trauma, i.e. 1.0.
+const TRAUMA_PER_PRESS: f32 = 0.4;
 
 fn main() {
     App::new()
         .add_plugins(DefaultPlugins)
         .add_systems(Startup, (setup_scene, setup_instructions, setup_camera))
-        .add_systems(Update, (screen_shake, trigger_shake_on_space))
+        // At the start of the frame, restore the camera's transform to its unshaken state.
+        .add_systems(PreUpdate, reset_transform)
+        .add_systems(
+            Update,
+            // Increase trauma when the space key is pressed.
+            increase_trauma.run_if(input_just_pressed(KeyCode::Space)),
+        )
+        // Just before the end of the frame, apply the shake.
+        .add_systems(PostUpdate, shake_camera.before(TransformSystems::Propagate))
```

Comment:
Curious to know if this ordering against `TransformSystems::Propagate` actually matters. If so, would be great if the comment explained a tiny bit more.

Reply:
Done: https://github.com/bevyengine/bevy/pull/19724/commits/22a5323c16b04e131046bf4c123a715cd85a7f61